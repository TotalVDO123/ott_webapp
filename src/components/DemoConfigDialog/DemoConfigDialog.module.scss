@use 'src/styles/variables';
@use 'src/styles/theme';

.controls {
  margin-top: variables.$base-spacing;

  > button {
    margin-right: 8px;
  }
}

.configModal {
  position: absolute;
  top: 0;
  width: 100%;
  height: 100%;
  background: var(--body-background-color);

  p {
    max-width: 400px;
    margin-bottom: 0;
    color: theme.$text-field-resting-color;
    font-size: 14px;
    line-height: 18px;
  }

  a {
    color: theme.$text-field-resting-color;
    font-weight: var(--body-font-weight-bold);
    text-decoration: underline;
  }
}

.note {
  padding: variables.$base-spacing;
  text-align: center;

  a {
    text-decoration: underline;
    cursor: pointer;
  }

  * {
    margin-top: calc(#{variables.$base-spacing} / 2);
    color: theme.$text-field-resting-color;
    font-weight: var(--body-font-weight-bold);
  }
}
