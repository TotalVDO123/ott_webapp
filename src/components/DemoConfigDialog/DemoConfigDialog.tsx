import React, { MouseEventHandler, useState } from 'react';
import { useTranslation } from 'react-i18next';

import styles from './DemoConfigDialog.module.scss';

import ErrorPage from '#src/components/ErrorPage/ErrorPage';
import TextField from '#src/components/TextField/TextField';
import Button from '#src/components/Button/Button';
import { addConfigQueryParam, clearStoredConfig, getConfigLocation } from '#src/utils/configOverride';
import Link from '#src/components/Link/Link';
import ConfirmationDialog from '#src/components/ConfirmationDialog/ConfirmationDialog';

const fallbackConfig = import.meta.env.APP_DEMO_DEFAULT_CONFIG_ID;

interface Props {
  configLocation?: string;
}

const DemoConfigDialog = ({ configLocation = getConfigLocation() }: Props) => {
  const { t } = useTranslation('demo');
  const [showDemoWarning, setShowWarning] = useState(false);

  const clearConfig = () => {
    clearStoredConfig();
    window.location.reload();
  };

  if (configLocation) {
    return (
      <div className={styles.note}>
        <div>{t('currently_previewing_config', { configLocation })}</div>
        <Link onClick={clearConfig}>{t('click_to_unselect_config')}</Link>
      </div>
    );
  }

  const cancelConfigClick: MouseEventHandler<HTMLButtonElement> = (event) => {
    event.preventDefault();

    if (fallbackConfig) {
      setShowWarning(true);
    }
  };

  const confirmDemoClick = () => {
    addConfigQueryParam(fallbackConfig);
    window.location.reload();
  };

  const cancelDemoClick = () => setShowWarning(false);

  return (
    <div className={styles.configModal}>
      <ErrorPage title={t('app_config_not_found')}>
        <form method="GET" action="/">
          <TextField required placeholder={t('please_enter_config_id')} name="app-config" />
          <div className={styles.controls}>
            <Button label={t('submit_config_id')} type={'submit'} />
            {fallbackConfig && <Button label={t('cancel_config_id')} onClick={cancelConfigClick} />}
          </div>
        </form>
        <p>
          {t('use_the_jwp_dashboard')}
          <span> </span>
          <Link href={'https://docs.jwplayer.com/platform/docs/ott-create-an-app-config'} target={'_blank'}>
            {t('learn_more')}
          </Link>
        </p>
      </ErrorPage>
      <ConfirmationDialog
        open={showDemoWarning}
        title={t('view_generic_demo')}
        body={t('viewing_config_demo')}
        onConfirm={confirmDemoClick}
        onClose={cancelDemoClick}
      />
    </div>
  );
};

export default DemoConfigDialog;
