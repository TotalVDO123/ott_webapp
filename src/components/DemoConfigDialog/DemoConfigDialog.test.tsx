import React from 'react';

import { renderWithRouter } from '#test/testUtils';
import DemoConfigDialog from '#src/components/DemoConfigDialog/DemoConfigDialog';

describe('<DemoConfigDialog>', () => {
  test('renders and matches snapshot', () => {
    const { container } = renderWithRouter(<DemoConfigDialog configLocation={''} />);

    expect(container).toMatchSnapshot();
  });

  test('renders and matches snapshot with ID', () => {
    const { container } = renderWithRouter(<DemoConfigDialog configLocation={'gnnuzabk'} />);

    expect(container).toMatchSnapshot();
  });
});
