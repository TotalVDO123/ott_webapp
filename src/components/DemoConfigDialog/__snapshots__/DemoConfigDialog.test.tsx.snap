// Vitest Snapshot v1

exports[`<DemoConfigDialog> > renders and matches snapshot 1`] = `
<div>
  <div
    class="_configModal_561a54"
  >
    <div
      class="_errorPage_8c5621"
    >
      <div
        class="_box_8c5621"
      >
        <header>
          <h1
            class="_title_8c5621"
          >
            app_config_not_found
          </h1>
        </header>
        <main
          class="_main_8c5621"
        >
          <form
            action="/"
            method="GET"
          >
            <div
              class="_textField_e16c1b"
            >
              <label
                class="_label_e16c1b"
                for="text-field_1235_app-config"
              />
              <div
                class="_container_e16c1b"
              >
                <input
                  class="_input_e16c1b"
                  id="text-field_1235_app-config"
                  name="app-config"
                  placeholder="please_enter_config_id"
                  required=""
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="_controls_561a54"
            >
              <button
                class="_button_f8f296 _default_f8f296 _outlined_f8f296"
                type="submit"
              >
                <span>
                  submit_config_id
                </span>
              </button>
            </div>
          </form>
          <p>
            use_the_jwp_dashboard
            <span>
               
            </span>
            <a
              class="_link_a14c64"
              href="https://docs.jwplayer.com/platform/docs/ott-create-an-app-config"
              rel="noreferrer"
              target="_blank"
            >
              learn_more
            </a>
          </p>
        </main>
      </div>
    </div>
  </div>
</div>
`;

exports[`<DemoConfigDialog> > renders and matches snapshot with ID 1`] = `
<div>
  <div
    class="_note_561a54"
  >
    <div>
      currently_previewing_config
    </div>
    <a
      class="_link_a14c64"
    >
      click_to_unselect_config
    </a>
  </div>
</div>
`;
